### PR TITLE
Add a --include-files flag

### PR DESCRIPTION
**Pull request by dawedawe**
- add a --include-files flag
- add globing support for analyzer filters
- rename filter flags for consistency

### CODE REVIEW

File: src/FSharp.Analyzers.SDK/FSharp.Analyzers.SDK.fsproj [hunk 1]
```diff
@@ -24,5 +24,6 @@
     <PackageReference Include="FSharp.Compiler.Service" />
     <PackageReference Include="McMaster.NETCore.Plugins" />
     <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" />
+    <PackageReference Include="Glob" />
```
Comment (nojaf) [inline src/FSharp.Analyzers.SDK/FSharp.Analyzers.SDK.fsproj:27]: Do we really want this to be inside the SDK?

Comment (dawedawe) [reply src/FSharp.Analyzers.SDK/FSharp.Analyzers.SDK.fsproj:27]: It's needed for the new globbing support of analyzers include/exclude lists.
Do you see a potential issue with it?

Comment (nojaf) [reply src/FSharp.Analyzers.SDK/FSharp.Analyzers.SDK.fsproj:27]: I was thinking of having that only in the cli tool.
Avoid as much as dependencies as possible for the analyzer authors.

Comment (dawedawe) [reply src/FSharp.Analyzers.SDK/FSharp.Analyzers.SDK.fsproj]: Okay, I went with a predicate parameter. That way, editor clients can still use globing if they want.

**Review by nojaf — APPROVED** [review]
Looks good!

**Pull request merged**